addons/project/models/res_partner.py

```diff
@@ -28,3 +31,22 @@ def _compute_task_count(self):
                 if partner in self:
                     partner.task_count += group['partner_id_count']
                 partner = partner.parent_id
+
+    def _create_portal_users(self):
+        partners_without_user = self.filtered(lambda partner: not partner.user_ids)
+        if not partners_without_user:
+            return self.env['res.users']
+        template_user_id = literal_eval(self.env['ir.config_parameter'].sudo().get_param('base.template_portal_user_id', 'False'))
+        User = self.env['res.users'].with_context(no_reset_password=True)
+        template_user = User.browse(template_user_id)
+        if not template_user.exists():
+            raise ValueError(_('Signup: invalid template user'))
+        portal_user_vals = template_user.copy_data({'active': True})[0]
```

Comment:
I'm guessing this tries to optimise the fetching of the copy data by only doing it once, but it seems to be rather dramatically broken when compared to the current version of `BaseModel.copy`, especially with regard to translation issues fixed by 428d9e81775dd470df6536ded4c6c56331efd760 and 60e9e1f738cb04e9d76fed05dda4caa1293963c4.

It also won't recursively copy inactive records (8606fb3df896389cc09986242da0e913a8515576) though that seems like a more minor concern for creating users from templates.

Unless this is a critically slow step of the workflow, I'd suggest just calling `copy`.

Reply:
Ok thanks for the explanation, I will use the [`_create_user_from_template` method in `res.users`](https://github.com/odoo/odoo/blob/c795927ce1e1c26b30ff1332def2318377828ccc/addons/auth_signup/models/res_users.py#L134-L152). 🙂